apps/condo/domains/meter/tasks/sendVerificationDateReminder.js

```diff
@@ -0,0 +1,241 @@
+     * Use DEFAULT_LOCALE if organization.country is unknown
+     * (not defined within @condo/domains/common/constants/countries)
+     */
+    return get(COUNTRIES, get(organization, 'country.locale'), DEFAULT_LOCALE)
+}
+const generateReminderMessages = async ({ context, reminderWindowSize, reminders }) => {
+    const messages = []
+
+    await Promise.all(reminders.map(async (reminder) => {
+        const { meter, residents } = reminder
+        const lang = await getOrganizationLang(context, meter.organization.id)
+
+        // prepare a message for each resident
+        messages.push(
+            ...residents.map(resident => ({
+                sender: { dv: 1, fingerprint: 'meters-validation-date-cron-push' },
+                to: { user: { id: resident.user.id } },
+                type: METER_VERIFICATION_DATE_REMINDER_TYPE,
+                lang,
+                meta: {
+                    dv: 1,
+                    data: {
+                        reminderWindowSize,
+                        reminderDate: dayjs(meter.nextVerificationDate).locale(lang).format('D MMM'),
+                        meterId: meter.id,
+                        userId: resident.user.id,
+                        residentId: resident.id,
+                    },
+                },
+            }))
+        )
+    }))
+
+    return messages
+}
+
+const sendReminders = async ({ context, messages }) => {
+    await Promise.all(messages.map(async (message) => {
+        await sendMessage(context, message)
+    }))
+}
+
+const sendVerificationDateReminder = async ({ date, searchWindowDaysShift, daysCount }) => {
+    // prepare vars
+    if (!date) date = dayjs().format('YYYY-MM-DD')
+    const reminderWindowSize = searchWindowDaysShift + daysCount
+
+    // initialize context stuff
+    const { keystone: context } = await getSchemaCtx('Meter')
+
+    // let's proceed meters page by page
+    const messagesQueue = []
+    const pageSize = 100
+    let offset = 0
+    let hasMore = false
+    do {
+        const meters = await readMetersPage({
+            context, offset, pageSize, date, searchWindowDaysShift, daysCount,
+        })
+
+        if (meters.length > 0) {
+            // connect residents to meter through the property field
+            const metersConnectedWithResidents = await getMetersConnectedWithResidents(
+                { context, meters }
+            )
+
+            if (metersConnectedWithResidents.length > 0) {
+                // filter out meters that was already reminded
+                const reminders = await filterSentReminders({
+                    context, date, reminderWindowSize, metersConnectedWithResidents,
+                })
+
+                // generate messages
+                const messages = await generateReminderMessages({
+                    context, reminderWindowSize, reminders,
+                })
+
+                // push to queue only unique messages
+                messages.forEach(message => {
```

Comment:
de we really need deduplication?

Reply:
I get rid of this. No reason for deduplication - was my mistake to introduce such functionality